From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "J. Nick Koston" <nick@cpanel.net>
Date: Wed, 29 Mar 2017 23:20:28 -0500
Subject: [PATCH 3/3] Replace using lsof with cpanel_dovecot_solr_isonline

case CPANEL-12281

This works around the fact that lsof does not work as a user
with cloudlinux ptrace block
---
 SOURCES/solr/bin/solr | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/SOURCES/solr/bin/solr b/SOURCES/solr/bin/solr
index b925d76..d2d4f2e 100755
--- a/SOURCES/solr/bin/solr
+++ b/SOURCES/solr/bin/solr
@@ -1709,13 +1709,13 @@ function launch_solr() {
 	1>"$SOLR_LOGS_DIR/solr-$SOLR_PORT-console.log" 2>&1 & echo $! > "$SOLR_PID_DIR/solr-$SOLR_PORT.pid"
 
     # no lsof on cygwin though
-    if hash lsof 2>/dev/null ; then  # hash returns true if lsof is on the path
+    if [ -x "/usr/local/cpanel/3rdparty/scripts/cpanel_dovecot_solr_isonline" ]; then
       echo -n "Waiting up to $SOLR_STOP_WAIT seconds to see Solr running on port $SOLR_PORT"
       # Launch in a subshell to show the spinner
       (loops=0
       while true
       do
-        running=`lsof -PniTCP:$SOLR_PORT -sTCP:LISTEN`
+        running=`/usr/local/cpanel/3rdparty/scripts/cpanel_dovecot_solr_isonline`
         if [ -z "$running" ]; then
 	  slept=$((loops * 2))
           if [ $slept -lt $SOLR_STOP_WAIT ]; then
@@ -1734,7 +1734,7 @@ function launch_solr() {
       done) &
       spinner $!
     else
-      echo -e "NOTE: Please install lsof as this script needs it to determine if Solr is listening on port $SOLR_PORT."
+      echo -e "NOTE: Please install /usr/local/cpanel/3rdparty/scripts/cpanel_dovecot_solr_isonline as this script needs it to determine if Solr is listening on port $SOLR_PORT."
       sleep 10
       SOLR_PID=`ps auxww | grep start\.jar | grep -w "\-Djetty\.port=$SOLR_PORT" | grep -v grep | awk '{print $2}' | sort -r`
       echo -e "\nStarted Solr server on port $SOLR_PORT (pid=$SOLR_PID). Happy searching!\n"
